Add docker.save make target

john-a-joyce opened the pull request:
This is needed by the release scripts to save the docker files so that the licenses can be added before pushing.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **john-a-joyce**

If they are not already assigned, you can assign the PR to them by writing `/assign @john-a-joyce` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/cni/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["john-a-joyce"]} -->

tiswanso commented on tools/istio-cni-docker.mk:
```diff
@@ -65,3 +66,26 @@ docker.push: $(DOCKER_PUSH_TARGETS)
 # This target will package all docker images used in test and release, without re-building
 # go binaries. It is intended for CI/CD systems where the build is done in separate job.
 docker.all: $(DOCKER_TARGETS)
+
```
Looks like much/most of this stuff a duplicate of the tools/istio-cni-docker.mk

tiswanso commented:
sorry for the noise comment... thought it was modifying the main Makefile

john-a-joyce commented:
Addressed review comments.   If this looks good can someone merge so i can fix up https://github.com/istio/istio/pull/9973.  I tested 9973 with my personal github and would like to change it to use master and restest.

istio-testing commented:
@venilnoronha: changing LGTM is restricted to assignees, and assigning you to the PR failed.

<details>

In response to [this](https://github.com/istio/cni/pull/68#pullrequestreview-176058492):

>/lgtm
>/approve


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/cni/pull/68#pullrequestreview-176058492" title="Approved">venilnoronha</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/istio/cni/blob/master/OWNERS)~~ [venilnoronha]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
